### PR TITLE
new lib_mail.sh

### PR DESCRIPTION
**Pull request by stefans73**
Mails are sent by function send_email in lib-mail.sh. The function send_email takes up to three arguments: subject, plain-text, html-text
If in config.cfg enabled, ip.sh sends a HTML-Link.

### CODE REVIEW

File: scripts/config.cfg [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+# Mail settings (specify to receive notifications)
+# If the mail settings are specified, Little Backup Box
+# sends an email with the device's IP  address even when
+# $NOTIFY is set to false
+SMTP_SERVER=""
+SMTP_PORT=""
+MAIL_USER=""
+MAIL_PASSWORD=""
+MAIL_TO=""
+MAIL_HTML="true" # Set to false to disable HTML-Mails
```
Comment (dmpop) [inline scripts/config.cfg:17]: I think that there is no need for double quotes: `MAIL_HTML=true` is correct

File: scripts/lib-mail.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    CONFIG_DIR=$(dirname "$0")
+    CONFIG="${CONFIG_DIR}/config.cfg"
+    dos2unix "$CONFIG"
+    source "$CONFIG"
+
+    BOUNDARY="${RANDOM}${RANDOM}${RANDOM}"
+    TEXT=""
+    
+    #Mail-body
+    if [ "${MAIL_HTML}" = "true" ];
```
Comment (dmpop) [inline scripts/lib-mail.sh:40]: Same here: `if [ "${MAIL_HTML}" = true ];`

**Review by dmpop — COMMENTED** [review]
Thank you very much for another great pull request! I only have two minor points.

**Comment by stefans73** [comment]
Hi,

I saved it as draft. It's not tested outside of ip.sh.

I will rework it later and integrate your changes.

Thank you,
Stefan

Am 9. Juli 2021 08:45:59 MESZ schrieb Dmitri Popov ***@***.***>:
***@***.*** commented on this pull request.
>
>Thank you very much for another great pull request! I only have two
>minor points.
>
>> +STORAGE_MOUNT_POINT="/media/storage" # Mount point of the storage
>device
>+SOURCE_DEV="sdb1" # Name of the source device
>+SOURCE_MOUNT_POINT="/media/source" # Mount point of the source device
>+POWER_OFF=true # Set to false to disable automatic power off after
>backup
>+LOG=false # Set to true to enable logging
>+NOTIFY=false # Set to true to enable email notifications
>+# Mail settings (specify to receive notifications)
>+# If the mail settings are specified, Little Backup Box
>+# sends an email with the device's IP  address even when
>+# $NOTIFY is set to false
>+SMTP_SERVER=""
>+SMTP_PORT=""
>+MAIL_USER=""
>+MAIL_PASSWORD=""
>+MAIL_TO=""
>+MAIL_HTML="true" # Set to false to disable HTML-Mails
>
>I think that there is no need for double quotes: `MAIL_HTML=true` is
>correct
>
>> +    # Arguments
>+    SUBJECT=$1
>+    TEXT_PLAIN=$2
>+    TEXT_HTML=$3
>+
>+    # Config
>+    CONFIG_DIR=$(dirname "$0")
>+    CONFIG="${CONFIG_DIR}/config.cfg"
>+    dos2unix "$CONFIG"
>+    source "$CONFIG"
>+
>+    BOUNDARY="${RANDOM}${RANDOM}${RANDOM}"
>+    TEXT=""
>+    
>+    #Mail-body
>+    if [ "${MAIL_HTML}" = "true" ];
>
>Same here: `if [ "${MAIL_HTML}" = true ];`
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/dmpop/little-backup-box/pull/83#pullrequestreview-702729446

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Pull request closed without merging**